datanode/sqlstore/ledger.go

```diff
@@ -160,6 +163,77 @@ func (ls *Ledger) Query(
 	return &res, pageInfo, nil
 }
 
+func (ls *Ledger) Export(
+	ctx context.Context,
+	partyID, assetID string,
+	dateRange entities.DateRange,
+	pagination entities.CursorPagination,
+) ([]byte, entities.PageInfo, error) {
+	if partyID == "" {
+		return nil, entities.PageInfo{}, ErrLedgerEntryExportForParty
+	}
+
+	if assetID == "" {
+		return nil, entities.PageInfo{}, ErrLedgerEntryExportForAsset
+	}
+
+	filter := &entities.LedgerEntryFilter{
+		SenderAccountFilter: entities.AccountFilter{
+			AssetID:  entities.AssetID(assetID),
+			PartyIDs: []entities.PartyID{entities.PartyID(partyID)},
+		},
+	}
+
+	aggregatedEntries, pageInfo, err := ls.Query(ctx, filter, dateRange, pagination)
+	if err != nil {
+		return nil, entities.PageInfo{}, err
+	}
+
+	// Gather all data into a buffer.
+	buf := new(bytes.Buffer)
+
+	columns := []string{
+		"VegaTime", "AssetId", "SenderPartyId", "SenderMarketId", "SenderAccountType",
+		"ReceiverPartyId", "ReceiverMarketId", "ReceiverAccountType", "Quantity", "TransferType",
+	}
+
+	w := csv.NewWriter(buf)
+	err = w.Write(columns)
+	if err != nil {
+		return nil, entities.PageInfo{}, fmt.Errorf("error writing columns data: %w", err)
+	}
+
+	if aggregatedEntries != nil {
+		for _, record := range *aggregatedEntries {
+			row := []string{
+				record.VegaTime.String(),
+				string(*record.AssetID),
+				*record.ToProto().SenderPartyId,
+				*record.ToProto().SenderMarketId,
+				record.SenderAccountType.String(),
+				*record.ToProto().ReceiverPartyId,
+				*record.ToProto().ReceiverMarketId,
+				record.ReceiverAccountType.String(),
+				record.Quantity.String(),
+				vega.TransferType(*record.TransferType).String(),
+			}
+
+			err = w.Write(row)
+			if err != nil {
+				return nil, entities.PageInfo{}, fmt.Errorf("failed to write records to filename: %w", err)
```

Comment:
I think the error message needs updating

Reply:
This has been updated - Thanks!